### ISSUE TITLE
Launching the app manually during update installation stops the update

### ISSUE THREAD

**Issue by pietrovismara** [Squirrel/Squirrel.Mac#266]
After closing the app and starting the installation the update takes about 15s, after which the app is relaunched automatically.

If I launch the app manually before the installation is completed the effect is that the update is stopped. 

Is this expected behavior, and if it is how can I show some progress indicator or even prevent the app from opening during the installation?

**Comment by aprilandjan** [comment]
@pietrovismara That is the problem we encountered. Did you found any solution for this?

**Comment by pietrovismara** [comment]
@aprilandjan sorry for the late reply, I haven't.

**Comment by aprilandjan** [comment]
> @aprilandjan sorry for the late reply, I haven't.

Thanks for your reply. I digged a lot and found that it is hard to solve that. Given up right now 😭

**Comment by MarshallOfSound** [comment]
Yes, this is expected — and as of electron/electron#36130 (being upstreamed here in #304) it's an explicit safety abort: ShipIt checks for running instances of the target right before the final rename and bails with `SQRLInstallerErrorAppStillRunning` rather than risk swapping a bundle out from under a live process.

The progress-indicator / prevent-launch-during-install part doesn't exist in Squirrel today. Apps that care typically drop a sentinel file in app-support before `quitAndInstall` and check it on launch. If someone wants that built in, please open a fresh issue for it.
